### ISSUE TITLE
[wishlist] action: Apply and provide git-diff

### ISSUE THREAD

**Issue by chrysn** [crate-ci/typos#1046]
When typos is used as the GitHub action, changes are shown but not easy to apply, requiring the tedious process of opening the files, changing the text and not missing one suggestion (guess who missed one again).

While there is the write_changes option, that requires extra steps to set up (not only once but at runtime – creating another PR in the process or whatnot), and moreover, the one linked action that'd do that has been archive.

Please consider adding an option `patch` that
* runs typos in write-mode
* shows the output of `git diff`

A developer applying changes from CI often has a terminal open anyway, so seeing that the typos linter triggered and fixing it would be as easy as typing `git am` and copy-pasting from the action output.

CC'ing @ROMemories who was so kind as to introduce me to this tool ;-)

**Comment by epage** [comment]
As a user, I am not a patch heavy person and would find this bloating the output with noisy text.

As a maintainer, I am not a patch heavy person and have the impression there are multiple, slightly incompatible patch formats (`git diff` vs `diff`) and wouldn't be thrilled with having to deal with those.

I say this not to say "no" but in looking for more input (and not just from you but anyone) on this.

**Comment by chrysn** [comment]
Two observations that I hope help in the discussion:
* GitHub action output can collapse verbose output sections if the right markers are in there.
* Even the default output of `git show` (with more metadata than the typo diff would have) is readily accepted both by patch and `dit am`; if non-GNU patch is less permissive, I can look into options to maximize compatibility.
